src/watcher/watcher.ts

```diff
@@ -64,12 +64,26 @@ class AppointmentStateReducer implements StateReducer<WatcherAppointmentState, B
     }
 }
 
+/**
+ * Returns true if and only if the given predicate is true on `(state, block)` and false on `(oldState, oldBlock)`.
+ * TODO:198: could simplify this by removing the block parameter, and changing anchor states accordingly
+ */
+function predicateBecameTrue<TState, TBlock>(
```

Comment:
You could put this on the Component<T> class

Reply:
Doesn't really work for MappedComponents, though, as the predicate is on single items, rather than on the whole state

Reply:
Perhaps I can just export it in component.ts as a utility function?

Reply:
Do we have a mappedcomponent, I thought you'd removed that?

Reply:
I removed the StandardMappedComponent, that also implemented handleNewStateEvents as a list of condition/action.
The MappedComponent just computes a reducer by mapping a uniform collection of object (string-indexed) to reducers for each object.

Reply:
Hmm, still not sure what you mean, I cant see a MappedComponent class on your branch.

Reply:
Oh, right, we only have the MappedStateReducer. But still, Watcher extends Watcher extends Component<MappedState<WatcherAppointmentState>, Block>, while the preedicate is on WatcherAppointmentState only.

Reply:
OK, I guess the next best place to put it is on the anchor state itself

Reply:
In fact, we can create methods that match the predicates in the handleStateEvent function. And just call these directly on the state.